vault/resource_pki_secret_backend_config_urls.go

```diff
@@ -0,0 +1,136 @@
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Description: "Specifies the URL values for the OCSP Servers field. This can be a comma-separated string list.",
+				ForceNew:    false,
+			},
+		},
+	}
+}
+
+func pkiSecretBackendConfigUrlsCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+
+	path := pkiSecretBackendConfigUrlsPath(backend)
+
+	issuingCertificates := d.Get("issuing_certificates").(string)
+	crlDistributionsPoints := d.Get("crl_distribution_points").(string)
+	ocspServers := d.Get("ocsp_servers").(string)
+
+	data := map[string]interface{}{
+		"issuing_certificates":    issuingCertificates,
+		"crl_distribution_points": crlDistributionsPoints,
+		"ocsp_servers":            ocspServers,
+	}
+
+	log.Printf("[DEBUG] Creating URL config on PKI secret backend %q", backend)
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("error creating URL config PKI secret backend %q: %s", backend, err)
+	}
+	log.Printf("[DEBUG] Created URL config on PKI secret backend %q", backend)
+
+	d.SetId(fmt.Sprintf("%s/config/urls", backend))
+	return pkiSecretBackendConfigUrlsRead(d, meta)
+}
+
+func pkiSecretBackendConfigUrlsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	backend := pkiSecretBackendConfigUrlsPath(path)
+
+	log.Printf("[DEBUG] Reading URL config from PKI secret backend %q", backend)
+	config, err := client.Logical().Read(path)
+
+	if err != nil {
+		log.Printf("[WARN] Removing path %q its ID is invalid", path)
+		d.SetId("")
+		return fmt.Errorf("invalid path ID %q: %s", path, err)
+	}
+
+	d.Set("issuing_certificates", config.Data["issuing_certificates"])
+	d.Set("crl_distribution_points", config.Data["crl_distribution_points"])
+	d.Set("ocsp_servers", config.Data["ocsp_servers"])
+
+	return nil
+}
+
+func pkiSecretBackendConfigUrlsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Id()
+
+	path := pkiSecretBackendConfigUrlsPath(backend)
+
+	issuingCertificates := d.Get("issuing_certificates").(string)
+	crlDistributionsPoints := d.Get("crl_distribution_points").(string)
+	ocspServers := d.Get("ocsp_servers").(string)
+
+	data := map[string]interface{}{
+		"issuing_certificates":    issuingCertificates,
+		"crl_distribution_points": crlDistributionsPoints,
+		"ocsp_servers":            ocspServers,
+	}
+
+	log.Printf("[DEBUG] Updating URL config on PKI secret backend %q", backend)
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("error updating URL config for PKI secret backend %q: %s", backend, err)
+	}
+	log.Printf("[DEBUG] Updated URL config on PKI secret backend %q", backend)
+
+	return pkiSecretBackendConfigUrlsRead(d, meta)
+
+}
+
+func pkiSecretBackendConfigUrlsDelete(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
Will the linked issue be solved if this isn't implemented?

Reply:
There is no delete method in the Vault API.

The linked issue uses a generic secret as a workaround, for which the Vault provider expects to have a delete method. Since Vault does not allow `DELETE /pki/config/urls`, a terraform destroy fails, leaving an orphaned mount behind. The only option to proceed is to manually disable the mounted secret. 

A workaround for the lack of delete method would be to update with empty strings. However, it might make more sense to just stick with the Vault API.

Reply:
Ah thanks for the explanation! Does that mean, then, that we would get the same behavior from the Terraform Vault Provider if we stripped having this PKI Delete method? Like if we just struck [this](https://github.com/terraform-providers/terraform-provider-vault/pull/399/files#diff-7a80efbc38ff34d692c9de568dc5080aR17) whole line?

Reply:
According to [Terraform's guide](https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources), the Delete method is mandatory for the resource to function.

Reply:
Gotcha! Thanks!